### ISSUE TITLE
Add support for ActionController instrumentation subscriptions

### ISSUE THREAD

**Issue by tlrdstd** [toptal/chewy#560]
`elasticsearch-rails` includes instrumentation which subscribes to the `ActiveSupport::Notifications` events and logs data to the Rails log: https://github.com/elastic/elasticsearch-rails/tree/master/elasticsearch-rails#activesupport-instrumentation

I have put together a patch to add support for this to Chewy by wholesale copying code from https://github.com/elastic/elasticsearch-rails/tree/master/elasticsearch-rails/lib/elasticsearch/rails/instrumentation and updating namespace/event names accordingly.

This patch lives at https://gist.github.com/tlrdstd/c5192939f6fdfab097d3551453a13202

If some version of this could be baked directly into Chewy, I think it would be really great for anyone who wants to see Elasticsearch query times in their Rails logs while using Chewy!

**Comment by pyromaniac** [comment]
Hmm, why can't you use this stuff directly from the `elasticsearch-rails` gem?

**Comment by tlrdstd** [comment]
Not really, because the event names used for `ActiveSupport::Notifications` are different between the two gems. `elasticsearch-rails` [fires](https://github.com/elastic/elasticsearch-rails/blob/master/elasticsearch-rails/lib/elasticsearch/rails/instrumentation/publishers.rb#L25) `search.elasticsearch`, while `chewy` [fires](https://github.com/toptal/chewy/blob/master/lib/chewy/search/request.rb#L955) `search_query.chewy` events.

So [my patch](https://gist.github.com/tlrdstd/c5192939f6fdfab097d3551453a13202#file-lib_chewy_support_instrumentation-rb-L30-L43) mostly just updates the `LogSubscriber` to listen for `search_query.chewy`.

I could theoretically include the `elasticsearch-rails` gem and write a subscriber similar to https://github.com/toptal/chewy#newrelic-integration to re-broadcast `search_query.chewy` events as `search.elasticsearch` events. However, I'm not 100% sure that kind of event chaining would have the desired effect. Even if it did work, I don't love including the entire `elasticsearch-rails` gem in my project just so I can `require 'elasticsearch/rails/instrumentation'`.

**Comment by pyromaniac** [comment]
Ok, I'll consider this, but I need to dig deeper and it will certainly be rather later than sooner.
